### PR TITLE
Further updates to ExampleComponent and CLI for standards

### PR DESCRIPTION
**Pull request by dcwarwick**
Contributes to #393

#### What did you change?

ExampleComponent and CLI templates and CODE_GUIDELINES.MD

#### How did you test and verify your work?

Ran a build and ran storybook.

### CODE REVIEW

File: packages/experimental/src/components/ExampleComponent/_example-component.scss [hunk 1]
```diff
@@ -21,11 +22,11 @@
     --#{$pkg-prefix}-border-color: transparent;
   }
 
-  .#{$block-class}--boxed-set {
+  .#{$block-class}.#{$block-class}--boxed-set {
```
Comment (lee-chase) [inline packages/experimental/src/components/ExampleComponent/_example-component.scss:25]: Ooops.

```suggestion
  .#{$block-class}--boxed-set {
```

File: packages/experimental/src/components/ExampleComponent/_example-component.scss [hunk 2]
```diff
@@ -21,11 +22,11 @@
...
     border: 10px solid var(--#{$pkg-prefix}-border-color);
   }
 
-  .#{$block-class}--shadow-set {
+  .#{$block-class}.#{$block-class}--shadow-set {
```
Comment (lee-chase) [inline packages/experimental/src/components/ExampleComponent/_example-component.scss:29]:
```suggestion
  .#{$block-class}--shadow-set {
```

File: packages/experimental/src/components/ExampleComponent/ExampleComponent.js [hunk 3]
```diff
@@ -162,6 +169,11 @@ ExampleComponent.propTypes = {
    * The size for the buttons ('default', 'small' or 'field').
    */
   size: PropTypes.oneOf(['default', 'small', 'field']),
+
+  /**
+   * Optional style settings for the containing node.
+   */
+  style: PropTypes.object,
```
Comment (lee-chase) [inline packages/experimental/src/components/ExampleComponent/ExampleComponent.js:176]: Does this need a default for ...style to work?

File: packages/experimental/src/components/ExampleComponent/ExampleComponent.js [hunk 4]
```diff
@@ -25,77 +25,84 @@ const componentName = 'ExampleComponent';
...
+      primaryButtonLabel,
+      primaryKind,
+      secondaryButtonLabel,
+      secondaryKind,
+      size,
+      style,
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
```
Comment (lee-chase) [inline packages/experimental/src/components/ExampleComponent/ExampleComponent.js:47]: Can we test adding a ref?

**Review by lee-chase — CHANGES_REQUESTED** [review]

File: packages/experimental/src/components/ExampleComponent/_example-component.scss (hunk 1 above)
Comment (dcwarwick) [reply packages/experimental/src/components/ExampleComponent/_example-component.scss:25]: This was your suggestion, in #441 option A, that all component SCSS should include block class in selector.

File: packages/experimental/src/components/ExampleComponent/_example-component.scss (hunk 2 above)
Comment (dcwarwick) [reply packages/experimental/src/components/ExampleComponent/_example-component.scss:29]: ditto

File: packages/experimental/src/components/ExampleComponent/ExampleComponent.js (hunk 3 above)
Comment (dcwarwick) [reply packages/experimental/src/components/ExampleComponent/ExampleComponent.js:176]: No. `...undefined` contributes nothing to an object literal, so `{...undefined} = {}` and `{...undefined, foo: 42} = {foo:42}`. As this is a useful way to handle unspecified props I've added commentary to clarify this in the ExampleComponent now.

File: packages/experimental/src/components/ExampleComponent/ExampleComponent.js (hunk 4 above)
Comment (dcwarwick) [reply packages/experimental/src/components/ExampleComponent/ExampleComponent.js:47]: Good catch. Added.

**Review by lee-chase — APPROVED** [review]

**Pull request merged**